Source/WebKit/UIProcess/API/libwpe/TouchGestureController.cpp

```diff
@@ -120,10 +123,32 @@ TouchGestureController::EventVariant TouchGestureController::handleEvent(const s
         case GesturedEvent::None:
             break;
         case GesturedEvent::Click:
+        {
+            bool generateClick = true;
+
+#if ENABLE(CONTEXT_MENUS)
+            generateClick = (touchPoint->time - m_start.time) < contextMenuThreshold;
+#endif
+
+            if (generateClick) {
+                m_gesturedEvent = GesturedEvent::None;
+
+                ClickEvent generatedEvent;
+                generatedEvent.event = {
+                    wpe_input_pointer_event_type_null, touchPoint->time, touchPoint->x, touchPoint->y,
+                    0, 0, 0,
+                };
+                return generatedEvent;
+            }
+
+            m_gesturedEvent = GesturedEvent::ContextMenu;
+            FALLTHROUGH;
+        }
+        case GesturedEvent::ContextMenu:
         {
             m_gesturedEvent = GesturedEvent::None;
```

Comment:
Oops, I just noticed this. Something's wrong here: don't make sense to assign and then immediately reassign.

Reply:
Fixed. Could you have another look?